Algorithms/Graph/BellmanFord.cs

```diff
@@ -0,0 +1,120 @@
+using System;
+using System.Collections.Generic;
+using DataStructures.Graph;
+
+namespace Algorithms.Graph
+{
+    /// <summary>
+    /// Bellman-Ford algorithm on directed weighted graph.
+    /// </summary>
+    /// <typeparam name="T">Generic type of data in the graph.</typeparam>
+    public class BellmanFord<T>
+    {
+        private readonly DirectedWeightedGraph<T> graph;
+        private readonly Dictionary<Vertex<T>, double> distances;
+        private readonly Dictionary<Vertex<T>, Vertex<T>?> predecessors;
+
+        public BellmanFord(DirectedWeightedGraph<T> graph)
+        {
+            this.graph = graph;
+            distances = new Dictionary<Vertex<T>, double>();
+            predecessors = new Dictionary<Vertex<T>, Vertex<T>?>();
+        }
+
+        /// <summary>
+        /// Runs the Bellman-Ford algorithm to find the shortest distances from the source vertex to all other vertices.
+        /// </summary>
+        /// <param name="sourceVertex">Source vertex for shortest path calculation.</param>
+        /// <returns>
+        /// A dictionary containing the shortest distances from the source vertex to all other vertices.
+        /// If a vertex is unreachable from the source, it will have a value of double.PositiveInfinity.
+        /// </returns>
+        public Dictionary<Vertex<T>, double> Run(Vertex<T> sourceVertex)
+        {
+            InitializeDistances(sourceVertex);
+            RelaxEdges();
+            CheckForNegativeCycles();
+            return distances;
```

Comment:
Please make
```
        private readonly DirectedWeightedGraph<T> graph;
        private readonly Dictionary<Vertex<T>, double> distances;
        private readonly Dictionary<Vertex<T>, Vertex<T>?> predecessors;
```
method parameters, then you won't have to worry about shared state when calling the `Run` method multiple times

Reply:
Hi, I have made the changes. Hence resolving the conversation.